### ISSUE TITLE
Improve security and automation of official binary releases

### ISSUE THREAD

**Issue by na--** [k6io/k6#1247]
As pointed out by [this forum post](https://community.k6.io/t/signature-verification-disabled-for-rpm-packages/305), currently our RPM packages are not signed :disappointed: 

I think that during the transition away from AppVeyor/CircleCI (#959), we should also make a substantial improvement (mostly in terms of security) and upgrade in how we build our packages:
- start properly signing them - both the Linux ones and the Windows ones (https://github.com/loadimpact/k6/issues/1034 - which we've also had other issues with the bintray hosting before...)
- maybe investigate if we can also make the bulds reproducible (as [discussed](https://github.com/loadimpact/k6/pull/1245#discussion_r346252753) yesterday, though that might have to wait for Go 1.14: https://github.com/golang/go/issues/35435
- build and deploy the github releases from a CI instead of manually via `build-release.sh`

**Comment by na--** [comment]
Something else we should do for every commit, not just for tagged releases: actually build the rpm/deb/msi/etc. packages and test them (in the appropriate OS containers)... The v0.26.0 build of the linux packages broke: https://circleci.com/gh/loadimpact/k6/8360 :man_facepalming: If we were building these on every commit, or at least once daily, we would have noticed and fixed the issue much sooner.

**Comment by imiric** [comment]
Since the `go-bin-{deb,rpm}` tools are giving us issues and they don't seem to be well maintained (last commit was over 2 years ago), we might want to consider switching to https://github.com/jordansissel/fpm .

I haven't used fpm personally, but have heard good things when used with Python projects. Presumably we should be able to use the directory input option for Go binaries, and would have more output options, including macOS .pkg files, so it's worth looking into.

**Comment by imiric** [comment]
There's another issue with Homebrew packaging on macOS. The current [k6 formula](https://github.com/Homebrew/homebrew-core/blob/0efecb6cd027e6b8791ca2d3b03c21c78c0a7c30/Formula/k6.rb) actually builds the binary, which currently doesn't have any version information or any custom build flags we use for the GitHub releases. It's also a problem if we change our build system, since it currently uses `dep` and in v0.28.0 we switched to Go modules. This type of change hopefully won't happen again, but we shouldn't have to maintain two separate builds.

Ideally the binary shouldn't be rebuilt for Homebrew, and the formula should just fetch the `*-mac.zip` package from our GitHub release, verify the checksum, and bundle the binary for Homebrew. From a quick look, [this seems possible](https://jonathanchang.org/blog/maintain-your-own-homebrew-repository-with-binary-bottles/) but it would be messy to setup if have to use Bintray for this, so not sure, some investigation is needed for this.

**Comment by na--** [comment]
This is becoming a bit of a forced priority because of https://jfrog.com/blog/into-the-sunset-bintray-jcenter-gocenter-and-chartcenter/

**Comment by imiric** [comment]
The scope of this issue is a bit fuzzy, but most of the things mentioned have now been addressed:
- We replaced Bintray with our own package repositories at https://dl.k6.io.
- DEB/RPM packages are signed with our GPG key, and MSI packages with our certificate.
- `build-release.sh` is still used, but is now part of CI and run automatically for new releases.

What's still pending:
- Reproducible builds.
- Building packages more frequently, which is now tracked in #1970.
- Homebrew issues...
- Signing of RPM repository metadata, as mentioned in [the same forum post](https://community.k6.io/t/signature-verification-disabled-for-rpm-packages/305). We missed this recently, but should sign the metadata as well and enable `repo_gpgcheck`.

I suggest closing this issue and opening new ones for specific things. @na-- @MStoykov WDYT?
